sdk/storage/AzureStorageBlob/Source/Transport/URLSessionTransferManager.swift

```diff
@@ -107,6 +102,14 @@ internal final class URLSessionTransferManager: NSObject, TransferManager, URLSe
         return transfers[index]
     }
 
+    func register(client: StorageBlobClient?, forRestorationId restorationId: String) {
+        clients.setObject(client, forKey: restorationId as NSString)
```

Comment:
Might be worth adding the extension shown here (https://nshipster.com/nshashtable-and-nsmaptable/) for adding regular subscripting support to NSMapTable.

Reply:
I considered that, but since I also have to always cast the key to NSString, I decided to encapsulate that into a function, `client(forRestorationId:)` instead. It seemed to keep the callsites pretty clean.

Reply:
If you did the subscript wrapper could you not cast the String to NSString there?